rasterio/windows.py

```diff
@@ -350,6 +350,42 @@ def window_index(window):
     return tuple(slice(*w) for w in window)
 
 
+def round_window_to_full_tiles(window=None, block_shapes=None):
+    """
+    Round window to include full expanse of intersecting tiles.
+
+    Parameters
+    ----------
+    window : a Window or window tuple
+        The input window.
+
+    block_shapes : tuple of block shapes
+        The input raster's block shape. All bands must have the same block/stripe structure
+
+    Returns
+    -------
+    Window
+    """
+    if len(set(block_shapes)) != 1:
+        raise ValueError('All bands must have the same block/stripe structure')
+
+    height_shape = block_shapes[0][0]
+    width_shape = block_shapes[0][1]
+
+    row_range = window[0]
+    col_range = window[1]
+
+    row_min = int(row_range[0] / height_shape) * height_shape
```

Comment:
@perrygeo I'm actually not seeing that behavior. In this example, it seems to push the negative values in the right direction. Can you tell me which raster file you used as `src` so  I can do an exact comparison?
```
In [23]: src = rasterio.open("tests/data/rgb1.tif")

In [24]: set(src.block_shapes)
Out[24]: {(6, 400)}

In [25]: src.shape
Out[25]: (400, 400)

In [26]: tuple(round_window_to_full_tiles(((-1, 391), (-1,405)), src.block_shapes))
Out[26]: ((-6, 396), (-400, 800))
```